File: WebFlux/SpringWebFluxJavaObjectsFirst/src/main/java/com/instil/model/Course.java
package com.instil.model;

public class Course {
    private String id;
    private String title;
    private CourseDifficulty difficulty;
    private int duration;

    public Course() {
    }

    public Course(String id, String title, CourseDifficulty difficulty, int duration) {
        this.id = id;
        this.title = title;
        this.difficulty = difficulty;
        this.duration = duration;
    }

    public String getId() {
        return id;
    }

    public void setId(String id) {
        this.id = id;
    }

    public String getTitle() {
        return title;
    }

    public void setTitle(String title) {
        this.title = title;
    }

    public CourseDifficulty getDifficulty() {
        return difficulty;
    }

    public void setDifficulty(CourseDifficulty difficulty) {
        this.difficulty = difficulty;
    }

    public int getDuration() {
        return duration;
    }

    public void setDuration(int duration) {
        this.duration = duration;
    }
}

File: WebFlux/ClientForSpringServices/src/main/kotlin/com/instil/reactive/client/model/dto/Course.kt
package com.instil.reactive.client.model.dto

import com.instil.reactive.client.model.CourseDifficulty

class Course(
    var id: String,
    var title: String,
    var difficulty: CourseDifficulty,
    var duration: Int
) {
    constructor() : this("", "", CourseDifficulty.BEGINNER, 0)

    override fun toString(): String {
        return "$id called $title"
    }
}
